### PR TITLE
chore: implementing some unit tests

### PR DESCRIPTION
**Pull request by ricardomaraschini**
This is an ongoing effort. This PR implements unit tests for both `defaults` and `ssh` packages. As I move forward I will be opening new PRs for different packages.

### CODE REVIEW

File: pkg/defaults/provider.go [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+// DefaultsProvider is an entity that provides default values used during
+// HelmVM installation.
+type DefaultsProvider struct {
+	Base string
+}
+
+// Init makes sure all the necessary directory exists on the system.
+func (d *DefaultsProvider) Init() {
+	if err := os.MkdirAll(d.K0sctlBinsSubDir(), 0755); err != nil {
+		panic(fmt.Errorf("unable to create basedir: %w", err))
```
Comment (danj-replicated) [inline pkg/defaults/provider.go:37]: I'm not a massive fan of using `panic()` like this when an end user is going to be seeing the messages. it's nicer to use something like `logrus.Fatalf()`

**Review by danj-replicated — CHANGES_REQUESTED** [review]
aside from a few issues with the UX of using `panic` for fatal errors, looks good 👍

File: pkg/defaults/provider.go (hunk 1 above)
Comment (ricardomaraschini) [reply pkg/defaults/provider.go:37]: Yeah, that makes sense. Let me refactor that to replace the calls by `logrus.Fatalf`.

Comment (ricardomaraschini) [reply pkg/defaults/provider.go]: @danj-replicated solved, can you please take a look ?

**Review by danj-replicated — APPROVED** [review]

**Pull request merged**